src/coreclr/ilasm/sha256.h

```diff
@@ -0,0 +1,115 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+//*****************************************************************************
+// sha256.h
+//
+
+//
+// contains implementation of sha256 hash algorithm
+//
+//*****************************************************************************
+#ifndef __sha256__h__
+#define __sha256__h__
+
+#ifdef _WIN32
+inline HRESULT Sha256Hash(BYTE* pSrc, DWORD srcSize, BYTE* pDst, DWORD dstSize)
+{
+    if (dstSize < 32)
```

Comment:
Should this check for exact match?

Taking output buffer, filling part of it, and not telling the caller how much of it was filled is odd. The typical patterns for output buffers are:
- Take output buffer and fill it completely, fail if it is not possible to fill it completely
- Take output buffer and return how much of it was filled